test/functional/p2p_leak.py

```diff
@@ -88,11 +92,24 @@ def on_version(self, message):
         self.send_message(msg_ping())
         self.send_message(msg_getaddr())
 
+
+class P2PVersionStore(P2PInterface):
+    version_received = None
+
+    def on_version(self, msg):
+        super().on_version(msg)
+        self.version_received = msg
+
+
 class P2PLeakTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
         self.extra_args = [['-banscore=' + str(banscore)]]
 
+    def setup_network(self):
+        self.disable_mocktime()
+        self.setup_nodes()
+
```

Comment:
Why is this needed? Are we missing another PR or something?

Reply:
That's because (unlike in btc) our tests use mocktime by default (so that blocks are generated starting with genesis time instead of current time). This test relies on `time` to get `time_begin` (and `time_end`) so `assert_greater_than_or_equal(ver.nTime, time_begin)` is going to fail without this patch.